src/App/Fossa/LicenseScanner.hs

```diff
@@ -303,31 +308,89 @@ licenseScanSourceUnit ::
   ( Has Diagnostics sig m
   , Has (Lift IO) sig m
   , Has StickyLogger sig m
+  , Has Logger sig m
   , Has Exec sig m
   , Has ReadFS sig m
   , Has FossaApiClient sig m
   ) =>
+  Bool ->
```

Comment:
Can we make this an enum? Bools as arguments feel very unclear, especially for public functions in other modules.
Suggestion:
```haskell
data LicenseScanMode
  = LicenseScanModeSkipPreviouslyScanned
  | LicenseScanModeFull
```
Or similar (and similarly refactor the API calls and any call sites).